### PR TITLE
Porting TextCNN model from Tensorflow to Torch

### PR DESCRIPTION
**Pull request by Shiva-sankaran**
## Description

Implemented TextCNN in Torch. Following up on the https://github.com/deepchem/deepchem/issues/2863 for the Tensorflow to PyTorch model conversion list.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/torch_models/text_cnn.py [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+    's': 33
+}
+
+
+class TextCNN(nn.Module):
+    """
+    Torch implementation of a simple 1D CNN network
+    """
+
+    def __init__(self,
```
Comment (rbharath) [inline deepchem/models/torch_models/text_cnn.py:53]: Type annotations

File: deepchem/models/torch_models/text_cnn.py [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+            self.linear2 = nn.Linear(in_features=200,
+                                     out_features=self.n_tasks * 2)
+        else:
+            self.linear2 = nn.Linear(in_features=200,
+                                     out_features=self.n_tasks * 1)
+        self.relu = nn.ReLU()
+        self.softmax = nn.Softmax(dim=2)
+        self.highway = layers.HighwayLayer(200)
+
+    def forward(self, input):
```
Comment (rbharath) [inline deepchem/models/torch_models/text_cnn.py:120]: Docstrings and types

File: deepchem/models/torch_models/text_cnn.py [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+
+    # Below functions were taken from DeepChem TextCNN tensorflow implementation
+
+    def default_generator(self,
+                          dataset,
+                          epochs=1,
+                          mode='fit',
+                          deterministic=True,
+                          pad_batches=True):
+        """Transfer smiles strings to fixed length integer vectors"""
```
Comment (rbharath) [inline deepchem/models/torch_models/text_cnn.py:252]: Usage examples

**Review by rbharath — COMMENTED** [review]
Let's split this into 2 PRs. Code itself looks good but just needs a bit more polish

File: deepchem/models/torch_models/text_cnn.py [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+
+    References
+    ----------
+    .. [1]  Guimaraes, Gabriel Lima, et al. "Objective-reinforced generative adversarial networks (ORGAN) for sequence generation models." arXiv preprint arXiv:1705.10843 (2017).
+    .. [2] Kim, Yoon. "Convolutional neural networks for sentence classification." arXiv preprint arXiv:1408.5882 (2014).
+    .. [3] Srivastava et al., "Training Very Deep Networks".https://arxiv.org/abs/1507.06228
+
+    Examples
+    --------
+    >>> import os
```
Comment (shreyasvinaya) [inline deepchem/models/torch_models/text_cnn.py:170]: is this required here?

File: deepchem/models/torch_models/text_cnn.py [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+                 n_tasks,
+                 char_dict,
+                 seq_length,
+                 n_embedding=75,
+                 kernel_sizes=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 15, 20],
+                 num_filters=[
+                     100, 200, 200, 200, 200, 100, 100, 100, 100, 100, 160, 160
+                 ],
+                 dropout=0.25,
+                 mode="classification",
```
Comment (shreyasvinaya) [inline deepchem/models/torch_models/text_cnn.py:188]: Type annotations missing

**Comment by Shiva-sankaran** [comment]
Addressing the changes in the first of two PRs #3841

**Comment by Shiva-sankaran** [comment]
The follow-up second PR #3877

**Pull request merged**